link/xdp.go

```diff
@@ -0,0 +1,72 @@
+package link
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+// XDPAttachTypeFlags represents the XDP attach type.
+type XDPAttachTypeFlags uint32
+
+const (
+	// AttachTypeXDP tries to link as native XDP but in case
+	// the driver does not support native XDP, it will automatically
+	// fall back to generic XDP (default).
+	AttachTypeXDP XDPAttachTypeFlags = 0
+	// AttachTypeXDPGeneric links XDP BPF program for drivers which do not yet
+	// support native XDP.
+	AttachTypeXDPGeneric XDPAttachTypeFlags = 1 << iota
+	// AttachTypeXDPDriver links XDP BPF program into the driver’s
+	// receive path.
+	AttachTypeXDPDriver
+	// AttachTypeXDPOffload offloads the entire XDP BPF program into hardware.
+	AttachTypeXDPOffload
+)
+
+type XDPOptions struct {
+	// Program must be an XDP BPF program.
+	Program *ebpf.Program
+	// IfName is the interface name to attach to.
+	IfName string
```

Comment:
I think this should be `Interface int`. Less convenient, but `net.Interfaces()` is notorious for doing lots of netlink work.

Alternatively: `*net.Interface`

Reply:
I just wanted to propose supporting interface name and index together. I pushed the code, what do you think?
I think it's better to give options to developer then they can choose which one is better / works for them.
```go
link.AttachXDP(link.XDPOptions{
    Program: prog,
    Interface: "eth0",
})
```
or   
```go
link.AttachXDP(link.XDPOptions{
    Program: prog,
    Interface: 2,
})
```